docs/cookbook/adding-a-new-administration-page.md

```diff
@@ -115,6 +114,78 @@ The event subscriber should be auto-discovered by Symfony, reconfiguring the men
 
 ![Dashboard admin menu after the modification](img/dashboard-menu-after.png)
 
+## Access control
+
+When adding a new administrator page, you need to decide which [administrator roles](../administration/admin-rights.md) should have access to it.
+For the new agenda, you will probably need to create a new role. Usually, a `VIEW`/`FULL` pair of roles is used, so let's follow the convention here as well. In `Roles.php`, add the new roles, define theirs translations, and put them into the hierarchy:
+
+```diff
+
+<?php
+
+declare(strict_types=1);
+
+namespace App\Model\Security;
+
++ use Override;
+use Shopsys\FrameworkBundle\Model\Security\Roles as BaseRoles;
+
+class Roles extends BaseRoles
+{
++    public const string ROLE_X_VIEW = 'ROLE_X_VIEW';
++    public const string ROLE_X_FULL = 'ROLE_X_FULL';
++
++    /**
++     * {@inheritdoc}
++     */
++    #[Override]
++    public function getAvailableAdministratorRolesGrid(): array
++    {
++         $grid = parent::getAvailableAdministratorRolesGrid();
++
++         $grid[] = [
++             static::ROLE_X_FULL => t('X.com feed - full'),
++             static::ROLE_X_VIEW => t('X.com feed - view'),
++         ];
++
++         return $grid;
++    }
++
++    /**
++     * {@inheritdoc}
++     */
++    #[Override]
++    public static function getRolesHierarchy(): array
++    {
++        $rolesHierarchy = parent::getRolesHierarchy();
++
++        $rolesHierarchy[static::ROLE_ALL][] = static::ROLE_X_FULL;
++        $rolesHierarchy[static::ROLE_ALL_VIEW][] = static::ROLE_X_VIEW;
++        $rolesHierarchy[static::ROLE_X_FULL] = [static::ROLE_X_VIEW];
++
++        return $rolesHierarchy;
```

Comment:
Please don't hate me :D 

What about making it more like a configuration? 
I'm thinking about something like 

```php
protected static function configureAdditionalRoles(): void {
    self::addRolePair(viewRole: static::ROLE_X_VIEW, fullRole: static::ROLE_X_FULL)
}
```

then it would be much more readable in my opinion. What do you think?

Reply:
That is a great idea, what about this? fd487c8c43

Reply:
love it 💚